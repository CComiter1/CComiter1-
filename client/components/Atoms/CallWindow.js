import React from 'react'
import { connect } from 'react-redux'

import { initCCP } from '../../store'

class CallWindow extends React.Component {
  constructor() {
    super()
    this.callRef = React.createRef()
  }
  componentDidMount() {
    setTimeout(() => {
      this.props.loadInitialData(this.callRef)
    }, 0)
  }

  render() {
    return (
      <div
        id="call-window"
        allow="microphone"
        ref={this.callRef}
        style={{ 
          backgroundColor: '#0099cc',
          height: '465px',
          width: '320px',
          minWidth: '320px',
          margin: '0 5px',
        }}
      />
    )
  }
}

const mapState = null
const mapDispatch = dispatch => ({
  loadInitialData(element) {
    dispatch(initCCP(element.current))
  },
})

export default connect(mapState, mapDispatch)(CallWindow)
