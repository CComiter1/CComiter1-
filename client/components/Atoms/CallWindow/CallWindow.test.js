import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import { CallWindow } from './CallWindow'

describe('CallWindow', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <CallWindow />,
    ).toJSON()
    expect(component).toMatchSnapshot()
  })
})
