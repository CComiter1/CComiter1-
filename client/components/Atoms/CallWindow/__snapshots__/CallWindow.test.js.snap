// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallWindow renders correctly 1`] = `
<div
  allow="microphone"
  id="call-window"
  style={
    Object {
      "height": "465px",
      "width": "320px",
    }
  }
/>
`;
